### PR TITLE
Fix unique key warning in collection header

### PR DESCRIPTION
**Pull request by jerabekjiri**
No Issue

A small fix to resolve the unique key warning in the CollectionHeader component, that has been bothering the console log for a while.
Screenshot from 2022-02-04 03-30-20

### CODE REVIEW

File: src/components/headers/collection-header.tsx [hunk 1]
```diff
@@ -166,7 +166,7 @@ export class CollectionHeader extends React.Component<IProps, IState> {
       noDependencies
         ? this.context.user.model_permissions.delete_collection && (
             <DropdownItem
-              key={1}
+              key='1'
```
Comment (himdel) [inline src/components/headers/collection-header.tsx:169]: good catch, but can we use more descriptive keys please :)

I'd suggest reusing the data-cy names for keys as well, delete-version-dropdown etc. seem descriptive enough :)

**Review by brumik — APPROVED** [review]
Cool, thanks

File: src/components/headers/collection-header.tsx [hunk 2]
```diff
@@ -190,16 +191,15 @@ export class CollectionHeader extends React.Component<IProps, IState> {
...
-            key='2'
-            onClick={() =>
-              this.openDeleteModalWithConfirm(collection.latest_version.version)
-            }
-          >
-            {t`Delete version ${collection.latest_version.version}`}
-          </DropdownItem>
-        </div>
+        <DropdownItem
+          data-cy='delete-version-dropdown'
```
Comment (himdel) [inline src/components/headers/collection-header.tsx:195]: the same  data-cy value is already used in the first DropdownItem here

```suggestion
          data-cy='delete-version-version'
```

**Comment by brumik** [comment]
I am merging this for my PR so I don't have to rebase twice.

**Comment by jerabekjiri** [comment]
@brumik sorry, I thought I already merged this, thanks :smiling_face_with_tear:

**Pull request merged**